### PR TITLE
feat: billing customer overrides

### PR DESCRIPTION
**Pull request by turip**
## Overview

This implementation allows us to specify the overrides per customer. Right now we are only overriding the workflow configuration part, later if providers are there we can add that to this code.

The API is able to yield a merged (Billing)Profile that we will be using for the invoice creation, where we are snapshotting the current settings.

Additional overrides will be required for the default profile changes when provider overrides are in place OM-927

### CODE REVIEW

File: openmeter/billing/adapter/customeroverride.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+			Entity: billing.EntityCustomer,
+			Err:    billing.ErrCustomerNotFound,
+		}
+	}
+
+	return mapCustomerOverrideFromDB(dbCustomerOverride), nil
+}
+
+func (r adapter) DeleteCustomerOverride(ctx context.Context, input billing.DeleteCustomerOverrideInput) error {
+	err := r.client().BillingCustomerOverride.Update().
```
Comment (hekike) [inline openmeter/billing/adapter/customeroverride.go:153]: should we check if it updates any?

File: openmeter/ent/schema/billing.go [hunk 2]
```diff
@@ -112,6 +114,98 @@ func (BillingWorkflowConfig) Edges() []ent.Edge {
...
+}
+
+func (BillingCustomerOverride) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("customer", Customer.Type).
+			Ref("billing_customer_override").
+			Field("customer_id").
+			Unique().
+			Required().
+			Immutable(),
```
Comment (hekike) [inline openmeter/ent/schema/billing.go:201]: do we need cascade?

**Review by hekike — APPROVED** [review]

File: openmeter/billing/adapter/customeroverride.go (hunk 1 above)
Comment (turip) [reply openmeter/billing/adapter/customeroverride.go:185]: No, as the service already checked that and we are in the same transaction, but changed the code.

File: openmeter/ent/schema/billing.go (hunk 2 above)
Comment (turip) [reply openmeter/ent/schema/billing.go:201]: That's defined at the To edge.

**Review by hekike — APPROVED** [review]

**Pull request merged**